### ISSUE TITLE
When discovering iOS devices: `ProcessException: Resource temporarily unavailable`

### ISSUE THREAD

**Issue by andrewkolos** [flutter/flutter#153776]
As of today (8/20), on 3.24.0: 221 reports from 102 unique clients (top 20ish crasher)

From flutter commands (looking at first 20 reports): `flutter daemon`

```
ProcessException: Resource temporarily unavailable  Command: /usr/bin/xcrun simctl list devices booted iOS --json
at _ProcessImpl._start(process_patch.dart:402)
at Process.start(process_patch.dart:38)
at _runNonInteractiveProcess(process_patch.dart:579)
at Process.run(process_patch.dart:49)
at LocalProcessManager.run(local_process_manager.dart:72)
at ErrorHandlingProcessManager.run.<anonymous closure>(error_handling_io.dart:669)
at _run(error_handling_io.dart:564)
at ErrorHandlingProcessManager.run(error_handling_io.dart:668)
at _DefaultProcessUtils.run(process.dart:312)
at SimControl._listBootedDevices(simulators.dart:162)
at SimControl.getConnectedDevices(simulators.dart:187)
at IOSSimulatorUtils.getAttachedDevices(simulators.dart:75)
at IOSSimulators.pollingGetDevices(simulators.dart:49)
at PollingDeviceDiscovery._initTimer.<anonymous closure>(device.dart:498)
at _rootRun(zone.dart:1391)
at _CustomZone.run(zone.dart:1301)
at _CustomZone.runGuarded(zone.dart:1209)
at _CustomZone.bindCallbackGuarded.<anonymous closure>(zone.dart:1249)
at _rootRun(zone.dart:1399)
at _CustomZone.run(zone.dart:1301)
at _CustomZone.bindCallback.<anonymous closure>(zone.dart:1233)
at Timer._createTimer.<anonymous closure>(timer_patch.dart:18)
at _Timer._runTimers(timer_impl.dart:398)
at _Timer._handleMessage(timer_impl.dart:429)
at _RawReceivePort._handleMessage(isolate_patch.dart:184)
```

**Comment by jmagman** [comment]
`ErrorHandlingProcessManager` could handle this error code more generically and suggest the command be run again?

**Comment by andrewkolos** [comment]
> `ErrorHandlingProcessManager` could handle this error code more generically and suggest the command be run again?

Doing some research around what `Resource temporarily unavailable` actually means, I think this is a good suggestion.

**Comment by andrewkolos** [comment]
I actually don't see this crash on 3.24.1? I might need to find a way to search the internal tool by GitHub issue ID

**Comment by jmagman** [comment]
https://github.com/flutter/flutter/pull/154306 seems like a fine change, though I don't think it's the right fix for this.  I think when devices are being discovered, if there's an error when getting the simulators (even a "temp" error) then the right thing would be to skip the simulators, not tool exit.

It looks like there may just be a missing check to see if the simulator command line is working:
https://github.com/flutter/flutter/blob/55af75d57bc702d2cbd43ceaf2d5fd71c8f85d74/packages/flutter_tools/lib/src/ios/simulators.dart#L186-L187
```diff
+   if (!_xcode.isSimctlInstalled) {
+     return <BootedSimDevice>[];
+   }
    final Map<String, Object?> devicesSection = await _listBootedDevices();
```
`isSimctlInstalled` will catch process errors:
https://github.com/flutter/flutter/blob/55af75d57bc702d2cbd43ceaf2d5fd71c8f85d74/packages/flutter_tools/lib/src/macos/xcode.dart#L189-L191

**Comment by andrewkolos** [comment]
> #154306 seems like a fine change, though I don't think it's the right fix for this. I think when devices are being discovered, if there's an error when getting the simulators (even a "temp" error) then the right thing would be to skip the simulators, not tool exit.

Makes sense. However, if we skip a discovery step for any reason, we should tell the user. Every now and then, users file issues in the class of "flutter won't discover my device", and these are very difficult to troubleshoot (many things can go wrong, high likelihood of user error, etc.).
